File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

// Prototype declarations
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
float index_calculation(int nb_letters, int nb_words, int nb_sentences);

int main(void)
{
    string text = get_string("Text: ");
    int nb_letters = count_letters(text);
    int nb_words = count_words(text);
    int nb_sentences = count_sentences(text);
    index_calculation(nb_letters, nb_words, nb_sentences);
}


// Function to count letters
int count_letters(string text)
{
    int count;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            count = count + 1 ;
            //printf("%i letters", count);
        }
        else
        {
            count = count + 0;
        }
    }
    //printf("%i letters", count);
    //printf("\n");
    return count;
}


// Function to count words
int count_words(string text)
{
    int count = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        //printf("Letter analyzed is: %c\n", text[i]);
        if (isspace(text[i]) && text[-1] != text[1])
        {
            count = count + 1;
            //printf("Counter: %i\n",count);
        }
        else if (text[i + 1] == '\0')
        {
            count = count + 1;
        }
        else
        {
            count = count + 0;
        }
    }
    //printf("%i words", count);
    //printf("\n");
    return count;
}

// Function to count sentences
int count_sentences(string text)
{
    int count = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        //printf("Letter analyzed is: %c\n", text[i]);
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            count = count + 1;
            //printf("Counter: %i\n",count);
        }
        else
        {
            count = count + 0;
        }
    }
    //printf("%i sentences", count);
    //printf("\n");
    return count;
}

float index_calculation(int nb_letters, int nb_words, int nb_sentences)
{
    float L = 0;
    L = (float)nb_letters / (float)nb_words * 100.0;
    //printf("L is: %f\n", L);

    float S = 0;
    S = (float)nb_sentences / (float)nb_words * 100.0;
    //printf("S is: %f\n", S);

    float index = 0;
    index = (0.0588 * L - 0.296 * S - 15.8);
    //printf("Index: %f\n", index);

    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        index = round(index);
        printf("Grade %i\n", (int)index);
    }
    return index;

}

File: sentimental-readability/readability.py
def main():

    text = input("Text: ")
    index_calculation(count_letters(text), count_words(text), count_sentences(text))

# Function to count number of words


def count_words(text):
    count = len(text.split())
    print(f"Word count : {count}")
    return count

# Function to count number of letters


def count_letters(text):
    count = 0
    for i in text:
        if i.isalpha():
            count += 1
    print(f"letter count : {count}")
    return count

# Function to count number of sentences


def count_sentences(text):
    count = 0
    for i in text:
        if i == "." or i == "?" or i == "!":
            count += 1
    print(f"sentences count : {count}")
    return count

# Algo to calculate the Coleman-Liau index


def index_calculation(nb_letters, nb_words, nb_sentences):
    L = 0
    S = 0
    index = 0
    L = nb_letters / nb_words * 100
    S = nb_sentences / nb_words * 100
    # Coleman-Liau index calculation
    index = (0.0588 * L - 0.296 * S - 15.8)
    # Return the grade number
    if (index >= 16):
        print("Grade 16+")
    elif (index < 1):
        print("Before Grade 1")
    else:
        index = round(index)
        print(f"Grade {index}")
    return index


if __name__ == "__main__":
    main()
